src/usr/ipmiext/ipmidcmi.C

```diff
@@ -45,20 +45,89 @@ namespace SENSOR
     errlHndl_t getUserPowerLimit( uint16_t &o_powerLimit, bool &o_limitActive )
     {
-        o_powerLimit = 0;
+        errlHndl_t l_err = NULL;
+        IPMI::completion_code cc = IPMI::CC_UNKBAD;
 
         // assume the limit has not been activated
         o_limitActive = false;
+        o_powerLimit = 0;
 
-        errlHndl_t l_err = NULL;
+        // Get DCMI Capabilities Information
+        size_t len = 2;
+        uint8_t* data = new uint8_t[len];
+        data[0] = 0xDC; // Group Extension Identification
+        data[1] = 0x01; // Selector - Supported DCMI Capabilities
+
+        l_err = IPMI::sendrecv(IPMI::get_dcmi_capability_info(), cc, len, data);
+        do
+        {
+            if (l_err != NULL)
+            {
+                TRACFCOMP(g_trac_ipmi,
+                          "Failed to send DCMI Capabilities Command to BMC");
+            }
+            else if (cc != IPMI::CC_OK)
+            {
+                TRACFCOMP(g_trac_ipmi,
+                          "Get DCMI Capabilities Command: "
+                          "bad completion code from BMC=0x%x",
+                          cc);
+
+                /*@
+                 * @errortype       ERRL_SEV_UNRECOVERABLE
+                 * @moduleid        IPMI::MOD_IPMIDCMI
+                 * @reasoncode      IPMI::RC_GET_DCMI_CAP_CMD_FAILED
+                 * @userdata1       BMC IPMI Completion code.
+                 * @devdesc         Request to get DCMI Capabilities information
+                 *                  failed
+                 * @custdesc        The DCMI Capabilities Info Command retrieve
+                 *                  data from the BMC has failed.
+                 *
+                 */
+
+                l_err = new ERRORLOG::ErrlEntry(
+                        ERRORLOG::ERRL_SEV_UNRECOVERABLE,
+                        IPMI::MOD_IPMIDCMI,
+                        IPMI::RC_GET_DCMI_CAP_CMD_FAILED,
+                        static_cast<uint64_t>(cc), 0, true);
+
+                l_err->collectTrace(IPMI_COMP_NAME);
+            }
+            else
+            {
+                // from the DCMI spec v1.5, Revision 1.0, August 23, 2011
+                // DCMI Capabilities Response Command Format:
+                // cc:       byte 1   completion code
+                // data[0]:  byte 2   0xDC
+                // data[1]:  byte 3   Major Version (01h)
+                // data[2]:  byte 4   Minor Version (05h)
+                // data[3]:  byte 5   Parameter Revision (02h)
+                // data[4]:  byte 6   Reserved
+                // data[5]:  byte 7   Platform capabilities
+                //           [7:1] Reserved
+                //           [0]   Power management
+                // data[6]:  byte 8   Manageability Access Capabilities
+                if (data[5] & 0x1)
+                {
+                    delete[] data;
+                    break;
+                }
+                TRACFCOMP(g_trac_ipmi,
```

Comment:
In the case when we know that power management is not supported, then it should skip the code to read the sensor and return: 0, false (power limit not active) That will prevent system from ending up in safe mode.

Reply:
Thanks for the review.
I agree with you, but I don't see the error in my code.
In this patch, I skip the code that requests the Power Limit from BMC, if Power Management is not supported.
Maybe I didn`t understand you.

However, the code looks confusing. I will rework the patch.
It would be better if I add changes to a separate function.